hack/update-codegen.sh

```diff
@@ -6,14 +6,12 @@ set -o pipefail
 
 SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")/.."
 
-if  ! ( command -v controller-gen > /dev/null ); then
-  # Need to take an unreleased version of controller-tools
-  # because of controller-tools bug 302
-  echo "controller-gen not found, installing sigs.k8s.io/controller-tools@83f6193..."
+if ( ! ( command -v controller-gen > /dev/null )  || test "$(controller-gen --version)" != "Version: v0.4.0" ); then
+  echo "controller-gen not found or out-of-date, installing sigs.k8s.io/controller-tools@v0.4.0..."
```

Comment:
if [[ $(controller-gen --version 2>/dev/null) != "Version: v0.4.0" ]]; then

?

also, "incorrect version", not "out of date", though do we really want to be blocking people from using v0.4.1+? Are we assuming API incompatibilities?

Reply:
I hate it when I do codegen (in any of its forms) and the output changes in ways I don't understand. So, I prefer to pin versions.

Reply:
What would be changed without this pinning exactly?

Reply:
@danielmellado if we don't pin a specific version, then people who run `hack/update-codegen.sh` might get different output depending on the version of the binary installed on their local machine.